testutils/src/main/java/com/microsoft/identity/internal/testutils/shadows/ShadowHttpClient.java

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.internal.testutils.shadows;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.microsoft.identity.common.internal.net.AbstractHttpClient;
+import com.microsoft.identity.common.internal.net.HttpClient;
+import com.microsoft.identity.common.internal.net.HttpResponse;
+import com.microsoft.identity.common.internal.net.UrlConnectionHttpClient;
+import com.microsoft.identity.internal.testutils.HttpRequestInterceptor;
+import com.microsoft.identity.internal.testutils.MockHttpClient;
+
+import org.robolectric.annotation.Implementation;
+import org.robolectric.annotation.Implements;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Map;
+
+/**
+ * Allows us to mock http request responses by shadowing the {@link HttpClient}.
+ *
+ * @see MockHttpClient - for setting up interceptors and mock responses
+ * @see HttpRequestInterceptor - an implementation for intercepting http requests.
+ */
+@Implements(AbstractHttpClient.class)
+public class ShadowHttpClient {
+
+    public HttpResponse method(@NonNull HttpClient.HttpMethod httpMethod,
```

Comment:
should this be named as send() instead of method()?

Reply:
You could go all Apache style and be sendWithMethod().  I prefer shorter names, but think send is too short.

Reply:
I think it needs to be named the same thing as what the original method is named....so if we want to change the name in the shadow then we probably need to first change the name in the original method

Reply:
But when I chose these names, it was generally because they align with: https://javaee.github.io/javaee-spec/javadocs/javax/ws/rs/client/SyncInvoker.html

Reply:
Interesting, when I first read the name it confused me as its not obvious from the name what the method does. Maybe its just me or because of my c# background